libnopegl/src/backends/vk/texture_vk.c

```diff
@@ -347,7 +347,11 @@ static VkResult texture_vk_init(struct texture *s, const struct texture_params *
         ngli_assert(0);
     }
 
-    const VkFormatFeatureFlags features = get_vk_format_features(s->params.usage);
+    VkFormatFeatureFlags features = get_vk_format_features(s->params.usage);
+    if (s->params.usage & NGLI_TEXTURE_USAGE_SAMPLED_BIT && (
+        s->params.min_filter != NGLI_FILTER_NEAREST ||
+        s->params.mag_filter != NGLI_FILTER_NEAREST))
+        features |= VK_FORMAT_FEATURE_SAMPLED_IMAGE_FILTER_LINEAR_BIT;
```

Comment:
not blocking but `get_vk_format_features` should probably get `params` instead of `params.usage`; right now nothing else uses `get_vk_format_features` so it doesn't matter, but I would guess that when it happens you'll have to duplicate that whole chunk again.

Reply:
Adjusted to:
``` c
+static VkFormatFeatureFlags get_vk_texture_format_features(const struct texture_params *params)
+{
+    VkFormatFeatureFlags features = get_vk_format_features(params->usage);
+
+    if (params->usage & NGLI_TEXTURE_USAGE_SAMPLED_BIT && (
+        params->min_filter != NGLI_FILTER_NEAREST ||
+        params->mag_filter != NGLI_FILTER_NEAREST))
+        features |= VK_FORMAT_FEATURE_SAMPLED_IMAGE_FILTER_LINEAR_BIT;
+
+    return features;
+}
+
 static int get_mipmap_levels(int32_t width, int32_t height)
 {
     int mipmap_levels = 1;
@@ -347,11 +359,7 @@ static VkResult texture_vk_init(struct texture *s, const struct texture_params *
         ngli_assert(0);
     }

-    VkFormatFeatureFlags features = get_vk_format_features(s->params.usage);
-    if (s->params.usage & NGLI_TEXTURE_USAGE_SAMPLED_BIT && (
-        s->params.min_filter != NGLI_FILTER_NEAREST ||
-        s->params.mag_filter != NGLI_FILTER_NEAREST))
-        features |= VK_FORMAT_FEATURE_SAMPLED_IMAGE_FILTER_LINEAR_BIT;
+    VkFormatFeatureFlags features = get_vk_texture_format_features(params);
```